### PR TITLE
 13284 - Bio-dedupe changes for correction flow

### PR DESCRIPTION
**Pull request by mamta-344**
Bio-dedupe changes for correction flow

### CODE REVIEW

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java [hunk 1]
```diff
@@ -419,7 +421,7 @@ private void postAbisIdentification(InternalRegistrationStatusDto registrationSt
 			registrationStatusDto.setSubStatusCode(StatusUtil.BIO_DEDUPE_POTENTIAL_MATCH.getCode());
 			registrationStatusDto.setLatestTransactionStatusCode(RegistrationTransactionStatusCode.FAILED.toString());
 			moduleId = PlatformSuccessMessages.RPR_BIO_METRIC_POTENTIAL_MATCH.getCode();
-			packetInfoManager.saveManualAdjudicationData(matchedRegIds, registrationStatusDto.getRegistrationId(),
+			packetInfoManager.saveManualAdjudicationData(matchedRegIdsList, registrationStatusDto.getRegistrationId(),
```
Comment (vishwa-vyom) [inline registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java:425]: why is this changed done where matchedRegIdsList is not removed with its own reg ids ?

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/spi/packetmanager/PacketInfoManager.java [hunk 2]
```diff
@@ -199,6 +199,15 @@ public void saveManualAdjudicationData(List<String> uniqueMatchedRefIds, String
...
 	public List<RegBioRefDto> getBioRefIdByRegId(String regId);
 
+	/**
+	 * Gets the bio ref ids list by bioRefId.
+	 *
+	 * @param bioRefId
+	 *            the bio ref id
+	 * @return all the bioRefIds dto
+	 */
+	public List<RegBioRefDto> getRegBioRefDataByBioRefId(List<String> bioRefId);
```
Comment (vishwa-vyom) [inline registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/spi/packetmanager/PacketInfoManager.java]: Since it takes in a list, I think this method should be renamed to getRegBioRefDataByBioRefIds

File: registration-processor/registration-processor-info-storage-service/src/main/java/io/mosip/registration/processor/packet/storage/entity/RegBioRefEntity.java [hunk 3]
```diff
@@ -33,6 +33,12 @@
 	@Column(name = "upd_dtimes")
 	private LocalDateTime updDtimes;
 
+	@Column(name = "reg_type")
```
Comment (vishwa-vyom) [inline registration-processor/registration-processor-info-storage-service/src/main/java/io/mosip/registration/processor/packet/storage/entity/RegBioRefEntity.java]: Is it possible to add these 2 properties to the RegBioRefPKEntity class rather ?

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: registration-processor/core-processor/registration-processor-abis-handler-stage/src/main/java/io/mosip/registration/processor/abis/handler/stage/AbisHandlerStage.java [hunk 4]
```diff
@@ -436,14 +436,16 @@ private void createIdentifyRequest(List<AbisQueueDetails> abisQueueDetails, Stri
 	 * @param regId    the reg id
 	 * @param bioRefId the bio ref id
 	 */
-	private void insertInBioRef(String regId, String bioRefId) {
+	private void insertInBioRef(String regId, String bioRefId, String regType, int iteration) {
```
[the remark this answers is not in the document]
Comment (mamta-344) [reply registration-processor/core-processor/registration-processor-abis-handler-stage/src/main/java/io/mosip/registration/processor/abis/handler/stage/AbisHandlerStage.java]: ok i will change

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/spi/packetmanager/PacketInfoManager.java (hunk 2 above)
Comment (mamta-344) [reply registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/spi/packetmanager/PacketInfoManager.java]: ok

File: registration-processor/registration-processor-info-storage-service/src/main/java/io/mosip/registration/processor/packet/storage/entity/RegBioRefEntity.java (hunk 3 above)
Comment (mamta-344) [reply registration-processor/registration-processor-info-storage-service/src/main/java/io/mosip/registration/processor/packet/storage/entity/RegBioRefEntity.java]: yes.
but we have change the primary key of table also.

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java (hunk 1 above)
Comment (mamta-344) [reply registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java:425]: here i changed the matchedRegIds type as Set (which having only unique regids, earlier getting repeated regids while testing correction packet) and other methods are expecting list so created list with same matchedRegIds  and forget to remove the own reg ids. 
Now i fixed this.

File: registration-processor/registration-processor-info-storage-service/src/main/java/io/mosip/registration/processor/packet/storage/entity/RegBioRefEntity.java (hunk 3 above)
Comment (mamta-344) [reply registration-processor/registration-processor-info-storage-service/src/main/java/io/mosip/registration/processor/packet/storage/entity/RegBioRefEntity.java]: Need to change the primary key of table. I did the coding chnages.

File: registration-processor/core-processor/registration-processor-abis-handler-stage/src/main/java/io/mosip/registration/processor/abis/handler/stage/AbisHandlerStage.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (vishwa-vyom) [reply registration-processor/core-processor/registration-processor-abis-handler-stage/src/main/java/io/mosip/registration/processor/abis/handler/stage/AbisHandlerStage.java]: All the method names like getRegIdByRefIdAndRegTypeAndIteration should also call it process not regtype

File: registration-processor/registration-processor-info-storage-service/src/main/java/io/mosip/registration/processor/packet/storage/entity/RegBioRefEntity.java (hunk 3 above)
Comment (vishwa-vyom) [reply registration-processor/registration-processor-info-storage-service/src/main/java/io/mosip/registration/processor/packet/storage/entity/RegBioRefEntity.java]: Remove the unwanted line RegBioRefEntity.java

File: registration-processor/core-processor/registration-processor-abis-handler-stage/src/main/java/io/mosip/registration/processor/abis/handler/stage/AbisHandlerStage.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (vishwa-vyom) [reply registration-processor/core-processor/registration-processor-abis-handler-stage/src/main/java/io/mosip/registration/processor/abis/handler/stage/AbisHandlerStage.java]: PacketInfoDao.java alone check the variable names

**Review by vishwa-vyom — APPROVED** [review]
